exp/views/study.py

```diff
@@ -29,9 +31,40 @@ class StudyListView(LoginRequiredMixin, generic.ListView):
     StudyListView shows a list of studies that a user has permission to.
     '''
     model = Study
+    template_name = 'studies/study_list.html'
 
     def get_queryset(self, *args, **kwargs):
-        return get_objects_for_user(self.request.user, 'studies.can_view')
+        request = self.request.GET
+        queryset = get_objects_for_user(self.request.user, 'studies.can_view')
+
+        state = request.get('state')
+        if state and state != 'all':
+            if state == 'myStudies':
+                queryset = queryset.filter(creator=self.request.user)
+            else:
+                queryset = queryset.filter(state=state)
+
+        match = request.get('match')
+        if match:
+            queryset = queryset.filter(Q(name__icontains=match) | Q(short_description__icontains=match))
+
+        sort = request.get('sort')
+        if sort:
+            if 'name' in sort:
+                queryset = queryset.order_by(sort)
+            elif 'beginDate' in sort:
+                queryset = sorted(queryset, key=lambda t: t.begin_date or timezone.now(), reverse=True if '-' in sort else False)
```

Comment:
This is probably fine for now. Once this list gets long it's going to use a lot of memory and be very slow. Sorting outside of SQL is a bit of an anti-pattern as it's going to evaluate the entire queryset in order to sort it. There's some SQL magic we can do that would allow you to find the beginning date and the end date in a subquery and then sort by that all inside of the database. I might add a TODO to optimize this and the one below it using a subquery or an annotation.